src/compiler/transformers/es2015.ts

```diff
@@ -4052,4 +4085,21 @@ namespace ts {
                 };
             })();`
     };
+
+    const templateObjectHelper: EmitHelper = {
+        name: "typescript:makeTemplateObject",
+        scoped: false,
+        priority: 0,
+        text: `
+            var __makeTemplateObject = (this && this.__makeTemplateObject) || function (cooked, raw) {

```

Comment:
What about:
```js
cooked.raw = raw;
if (Object.freeze) Object.freeze(cooked);
return cooked;
```

Or, the one-line alternative:
```js
return Object.defineProperty ? Object.defineProperty(cooked, "raw", { value: raw }) : (cooked.raw = raw, cooked);
```

Reply:
Uglify already produces a shorter output given this current implementation.

Reply:
_Slightly_ better one-liner (two less characters, just as readable and correct):
```ts
return Object.defineProperty ? Object.defineProperty(cooked, "raw", { value: raw }) : cooked.raw = raw, cooked;
```
Which is, coincidentally, how uglify restructures what daniel has right now when it minifies it.